app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -364,6 +364,9 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                 systemUiVisibilityHelper.onKeyboardVisibilityChanged(false);
                 hider.fixScroll();
                 return false;
+            } else if (actionId == android.R.id.closeButton) {
+                displayKissBar(false);
+                return true;
```

Comment:
New condition checking for `actionId == android.R.id.closeButton` second time is hard to read/understand.
Can you refactor this somehow, maybe use nested `if (isAllowHideKeyboardEnabled()) {} else {}`

Also `mPopup.dismiss` is not called anymore when `isAllowHideKeyboardEnabled` is false. This should be done always to get rid of some dialogs that still may be shown